public/pages/CreateDataStream/containers/CreateDataStream/CreateDataStream.tsx

```diff
@@ -55,8 +75,14 @@ export default class CreateDataStream extends Component<CreateDataStreamProps> {
           dataStream={this.dataStream}
           onCancel={this.onCancel}
           onSubmitSuccess={() => this.props.history.push(ROUTES.DATA_STREAMS)}
+          key={this.props.dataSourceId}
```

Comment:
All the input fields will be cleared once data source id changes, is that acceptable?

Reply:
I think so, at least with the current requirements. Will discuss more with the team and circle back.